src/utils/telemetry-openkit.ts

```diff
@@ -0,0 +1,201 @@
+      action.reportEvent('server_started');
+      action.reportValue('version', require('../../package.json').version);
+      action.reportValue('node_version', process.version);
+      action.reportValue('platform', process.platform);
+      action.leaveAction();
+    } catch (error) {
+      console.warn('Failed to track server start:', error);
+    }
+  }
+
+  /**
+   * Track Tool Usage
+   * @param toolName name of the tool
+   * @param success whether or not the tool call was successful
+   * @param duration duration of the tool call
+   * @returns
+   */
+  async trackToolUsage(toolName: string, success: boolean, duration?: number): Promise<void> {
+    if (!this.isEnabled) return;
+
+    await this.initPromise;
+    if (!this.session) return;
+
+    try {
+      const action = this.session.enterAction(`tool_${toolName}`);
+      action.reportEvent(success ? 'tool_success' : 'tool_error');
+      action.reportValue('tool_name', toolName);
+      action.reportValue('success', success ? 'true' : 'false');
+
+      if (duration !== undefined) {
+        action.reportValue('duration_ms', duration);
+      }
+
+      action.leaveAction();
+    } catch (error) {
+      console.warn('Failed to track tool usage:', error);
+    }
+  }
+
+  /**
+   * Track Errors
+   * @param error error message to be tracked
+   * @param context
+   * @returns
+   */
+  async trackError(error: Error, context?: string): Promise<void> {
+    if (!this.isEnabled) return;
+
+    await this.initPromise;
+    if (!this.session) return;
+
+    try {
+      const action = this.session.enterAction('error_occurred');
+      // reportError expects name and code, so we'll use error name and a generic error code
+      action.reportError(error.name || 'Error', 500);
+
+      if (context) {
+        action.reportValue('error_context', context);
+      }
+
+      action.reportValue('error_message', error.message);
+      if (error.stack) {
+        action.reportValue('error_stack', error.stack.substring(0, 1000)); // Limit stack trace length
+      }
+      action.leaveAction();
+    } catch (trackingError) {
+      console.warn('Failed to track error:', trackingError);
+    }
+  }
+
+  async shutdown(): Promise<void> {
+    if (!this.isEnabled) return;
+
+    await this.initPromise;
+
+    try {
+      if (this.session) {
+        this.session.end();
+      }
+      if (this.openKit) {
+        await new Promise<void>((resolve) => {
+          this.openKit!.shutdown(() => resolve());
+        });
+      }
+    } catch (error) {
+      console.warn('Failed to shutdown usage tracking:', error);
+    }
+  }
+}
+
+class NoOpTelemetry implements Telemetry {
+  async trackServerStart(): Promise<void> {}
+  async trackToolUsage(): Promise<void> {}
+  async trackError(): Promise<void> {}
+  async shutdown(): Promise<void> {}
+}
+
+export function createTelemetry(): Telemetry {
+  const isEnabled = process.env.DT_MCP_DISABLE_TELEMETRY !== 'true';
```

Comment:
This check is also inside the constructor of `DynatraceMcpTelemetry` - is this intentional?